pylot/planning/waypoint_planning_operator.py

```diff
@@ -0,0 +1,78 @@
+from collections import deque
+
+import carla
+
+from erdos.op import Op
+from erdos.utils import setup_csv_logging, setup_logging
+
+from pylot.map.hd_map import HDMap
+import pylot.planning.cost_functions
+from pylot.planning.messages import WaypointsMessage
+from pylot.simulation.carla_utils import get_map, to_carla_location
+from pylot.planning.utils import get_distance, get_waypoint_vector_and_angle
+
+import pylot.utils
+
+
+class WaypointPlanningOperator(Op):
```

Comment:
This seems to be a stripped down version of the PlanningOperator. Are there cases when we would want to use WaypointPlanningOperator over PlanningOperator? Is it required for the MPCAgentOperator?

Reply:
From my understanding, PlanningOperator will return a single waypoint for PID controller to follow, while the WaypointPlanningOperator returns a list of waypoints for MPC (+ PID) to follow. I considered re-using the PlanningOperator's output and interpolating the waypoints between the current ego position and the output-ed waypoint, but I figured providing all of the waypoints in this new operator would be a cleaner approach. 

Let me know if this is okay with you.

Reply:
Can we do this by checking the mpc_agent flag at runtime, and passing more waypoints downstream?

Reply:
Yeah sure, I'll do that!